From ed5eba0ff09b61e46ceb2a4d78602a9e795a0e25 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ilpo=20J=C3=A4rvinen?= <ilpo.jarvinen@helsinki.fi>
Date: Tue, 2 May 2017 23:54:02 +0300
Subject: [PATCH 4/5] Separate menuitem creators

These changes are need for the menu filtering that will be
added in a subsequent patch. A separate function for qubes
manager is to safely allow multiple calls to
qubes.make_menu_with_filter(...) without adding the manager
menuitem to each of them (one per each submenu).
---
 lib/qubes.lua.in | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/lib/qubes.lua.in b/lib/qubes.lua.in
index 5598958..428a2df 100644
--- a/lib/qubes.lua.in
+++ b/lib/qubes.lua.in
@@ -154,11 +154,20 @@ function qubes.get_colour_focus(c)
     return qubes.get_label(c).colour_focus
 end
 
+function qubes.manager_menuitem()
+    return {'Qubes Manager', 'qubes-manager',
+        '/usr/share/icons/hicolor/16x16/apps/qubes-logo-icon.png'}
+end
+
+function qubes.make_menuitem(vm, program)
+    return {program.Name, program.cmdline, program.icon_path}
+end
+
 function qubes.make_vm_menu(vm)
     local menu = {}
 
     for _, program in ipairs(menubar.utils.parse_dir(vm.path .. '/apps')) do
-        table.insert(menu, {program.Name, program.cmdline, program.icon_path})
+        table.insert(menu, qubes.make_menuitem(vm, program))
     end
     return {vm.name, menu, vm.path .. '/icon.png'}
 end
@@ -181,8 +190,7 @@ print('\n'.join('{} {} {}'.format(vm.name, vm.type, vm.dir_path)
         io.stderr:write('line=' .. line .. '\n')
         table.insert(menu, qubes.make_vm_menu(vm))
     end
-    table.insert(menu, {'Qubes Manager', 'qubes-manager',
-        '/usr/share/icons/hicolor/16x16/apps/qubes-logo-icon.png'})
+    table.insert(menu, qubes.manager_menuitem())
     return menu
 end
 
-- 
2.7.4

